packages/api/src/EmbeddedChatApi.ts

```diff
@@ -1124,4 +1124,21 @@ export default class EmbeddedChatApi {
     const data = response.json();
     return data;
   }
+
+  async userData(username: string) {
```

Comment:
I believe the function to fetch `userDetails` already exists, if I remember correctly. Previously, clicking on the user avatar would display the user details in the sidepane. Is there a specific reason why this is needed again?

Reply:
On clicking the user avatar we get access to the `user's _id ` but the similar thing is not available on clicking the mention. I could only get access to the user's `username` . Also the rocket chat's implementation of user info modal utilizes the api call via username, so i decided to go with the same approach. Thus , now we have two api's (1) which allows fetching userInfo through _id (2) allows fetching info through username

Reply:
Okay cool, understood